packages/world/ts/config/v2/world.ts

```diff
@@ -29,8 +29,9 @@ export type validateWorld<world> = {
       : key extends "enums"
         ? narrow<world[key]>
         : key extends "namespaces"
-          ? // ? validateNamespaces<world[key], extendedScope<world>>
-            ErrorMessage<`Namespaces config will be enabled soon.`>
+          ? world extends { namespace?: unknown; tables?: unknown; systems?: unknown }
```

Comment:
would like to revisit this pattern soon and refactor some things so is easier to maintain and doesn't become a place for things to fall through cracks as we expand on what can be in a namespace

thinking something like

```ts
// in store
type NamespaceInput = { tables?: TablesInput };
type SingleNamespaceInput = { namespace?: string } & NamespaceInput;
type StoreInput = SingleNamespaceInput & { codegen?: CodegenInput, deploy?: DeployInput };

// in world
type NamespaceInput = StoreNamespaceInput & { systems?: SystemsInput };
type SingleNamespaceInput = { namespace?: string } & NamespaceInput;
type WorldInput = SingleNamespaceInput & { namespaces?: NamespacesInput, codegen?: CodegenInput, deploy?: DeployInput };
```

then this validation can be something like
```ts
world extends SingleNamespaceInput ? ErrorMessage<...> : validateNamespaces<...>
```

Reply:
working on this here: https://github.com/latticexyz/mud/pull/2898